File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/fullscreen/FullScreenActivityInterceptor.java
package com.raxdenstudios.square.interceptor.commons.fullscreen;

import android.os.Bundle;
import android.view.WindowManager;

import com.raxdenstudios.square.interceptor.ActivitySimpleInterceptor;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;

/**
 * Created by Ángel Gómez on 26/12/2016.
 */
public class FullScreenActivityInterceptor extends ActivitySimpleInterceptor implements FullScreenInterceptor {

    public FullScreenActivityInterceptor(@NonNull FragmentActivity activity) {
        super(activity);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int flags = WindowManager.LayoutParams.FLAG_FULLSCREEN;
        int mask = WindowManager.LayoutParams.FLAG_FULLSCREEN;

        mActivity.getWindow().setFlags(flags, mask);
    }

}

File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/fullscreen/FullScreenActivityInterceptor.kt
package com.raxdenstudios.square.interceptor.commons.fullscreen

import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.view.WindowManager

import com.raxdenstudios.square.interceptor.ActivitySimpleInterceptor

/**
 * Created by Ángel Gómez on 26/12/2016.
 */
class FullScreenActivityInterceptor(activity: FragmentActivity)
    : ActivitySimpleInterceptor(activity),
        FullScreenInterceptor {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activity.window.setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
        )
    }

}
